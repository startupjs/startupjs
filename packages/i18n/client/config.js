import cloneDeep from 'lodash/cloneDeep'
import defaults from 'lodash/defaults'

const DEFAULT_LANGUAGE = 'en'
let localConfig = {}

export function initConfig (config) {
  if (Object.keys(localConfig).length) {
    throw new Error(
      '[@dmapper/orgs] initConfig: config already initialized'
    )
  }

  Object.assign(localConfig, getLocalConfig(config))
}

function getLocalConfig (baseConfig) {
  const config = defaults(
    cloneDeep(baseConfig),
    { lang: DEFAULT_LANGUAGE, langs: [] }
  )
  const getLangs = config.getLangs

  if (typeof getLangs === 'function') {
    let langs = getLangs()

    if (!Array.isArray(langs)) {
      throw new Error(
        '[@staratupjs/i18n]: getLangs ' +
        'must return an array of languages'
      )
    }

    delete config.getLangs
    config.langs = langs
  }

  const { lang, langs } = config

  if (!langs.includes(lang)) {
    config.langs = [lang].concat(langs)
  }

  return config
}

export function useConfig () {
  return localConfig
}

export function getConfig () {
  return localConfig
}
