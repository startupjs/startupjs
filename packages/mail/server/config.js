const _config = {}

export default _config
