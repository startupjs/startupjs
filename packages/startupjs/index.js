export * from '@startupjs/react-sharedb'
export * from '@startupjs/hooks'
// HINT: `isomorphic` means that the code can be executed both
//        on the server and on the client
export * from '@startupjs/isomorphic-helpers'
// dummy babel macro functions for @startupjs/babel-plugin-rn-stylename-inline
export function css () {}
export function styl () {}
export { default as t } from '@startupjs/i18n/client/t.js'
