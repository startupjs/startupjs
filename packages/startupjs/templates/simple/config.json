{
  "NODE_ENV": "development",
  "PORT": 3000,
  "SESSION_SECRET": "REPLACE_SESSION_SECRET",
  "BASE_URL": "http://localhost:3000",
  "REDIS_URL": "redis://localhost:6379/1",
  "MONGO_URL": "mongodb://localhost:27017/helloworld",
  "COPY_TO_ENV": ["MONGO_URL"]
}
