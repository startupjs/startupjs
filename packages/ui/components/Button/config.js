const { u } = require('./../../config/helpers')

module.exports = function ({ colors }) {
  return {
    heights: {
      xs: u(2),
      s: u(3),
      m: u(4),
      l: u(5),
      xl: u(6),
      xxl: u(7)
    },
    outlinedBorderWidth: 1,
    iconMargins: {
      xs: u(0.5),
      s: u(0.5),
      m: u(1),
      l: u(1),
      xl: u(1.5),
      xxl: u(1.5)
    }
  }
}
