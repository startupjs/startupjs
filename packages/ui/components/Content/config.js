module.exports = function ({ colors }) {
  return {
    defaultWidth: 'mobile'
  }
}
