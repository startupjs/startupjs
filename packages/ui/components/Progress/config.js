module.exports = function () {
  return {
    duration: 300
  }
}
