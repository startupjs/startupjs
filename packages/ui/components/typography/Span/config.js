module.exports = function ({ colors }) {
  return {
    color: colors.mainText,
    descriptionColor: colors.secondaryText
  }
}
