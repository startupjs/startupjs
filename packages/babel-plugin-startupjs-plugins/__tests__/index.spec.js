const { readFileSync } = require('fs')
const { join } = require('path')
const pluginTester = require('babel-plugin-tester').default
const plugin = require('../index')
const { name: pluginName } = require('../package.json')

const FIXTURES_PATH = join(__dirname, '../fixtures')

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  pluginOptions: {
    root: FIXTURES_PATH
  },
  babelOptions: {
    filename: join(FIXTURES_PATH, './node_modules/config/index.js')
  },
  tests: {
    'Ignores files without a magic import': /* js */`
      import config from './startupjs.config.js'
      console.log(config)
    `,
    'Processes files with a magic import': /* js */`
      import { registry } from 'startupjs/registry'
      import config from './startupjs.config.virtual.js'
      import models from './startupjs.models.virtual.js'
      import features from './startupjs.features.virtual.js'
      import plugins from './startupjs.plugins.virtual.js'
      import dummy from '@dummy/dummy'

      config.features = features
      registry.init(config, { plugins, models })

      const x = 'xxx'
      dummy(x)

      export default () => {}
    `,
    'Test sample file from fixtures which loads config':
      readFileSync(join(FIXTURES_PATH, './node_modules/config/index.js'), 'utf8')
  }
})

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  pluginOptions: {
    root: FIXTURES_PATH,
    useRequireContext: true
  },
  babelOptions: {
    filename: join(FIXTURES_PATH, './node_modules/config/index.js')
  },
  tests: {
    'On Metro uses require.context for models': /* js */`
      import { registry } from 'startupjs/registry'
      import config from './startupjs.config.virtual.js'
      import models from './startupjs.models.virtual.js'
      import features from './startupjs.features.virtual.js'
      import plugins from './startupjs.plugins.virtual.js'
      import dummy from '@dummy/dummy'

      config.features = features
      registry.init(config, { plugins, models })

      const x = 'xxx'
      dummy(x)

      export default () => {}
    `
  }
})
