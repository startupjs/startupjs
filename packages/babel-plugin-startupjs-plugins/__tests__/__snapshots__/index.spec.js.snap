// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@startupjs/babel-plugin-startupjs-plugins Ignores files without a magic import: Ignores files without a magic import 1`] = `

import config from './startupjs.config.js'
console.log(config)

      ↓ ↓ ↓ ↓ ↓ ↓

import config from "./startupjs.config.js";
console.log(config);


`;

exports[`@startupjs/babel-plugin-startupjs-plugins On Metro uses require.context for models: On Metro uses require.context for models 1`] = `

import { registry } from 'startupjs/registry'
import config from './startupjs.config.virtual.js'
import models from './startupjs.models.virtual.js'
import features from './startupjs.features.virtual.js'
import plugins from './startupjs.plugins.virtual.js'
import dummy from '@dummy/dummy'

config.features = features
registry.init(config, { plugins, models })

const x = 'xxx'
dummy(x)

export default () => {}

      ↓ ↓ ↓ ↓ ↓ ↓

import { registry } from "startupjs/registry";
import config from "../../startupjs.config.js";
import dummy from "@dummy/dummy";
import _default from "@startupjs/server/plugins/clientSession.plugin";
import _default2 from "@startupjs/ui/plugin";
import _default3 from "startupjs/plugins/cssMediaUpdater.plugin";
import _default4 from "module-1/plugin";
import _default5 from "module-1-plugin/thePlugin.plugin";
import _default6 from "../../dummyPlugin.plugin.js";
const plugins = [
  _default,
  _default2,
  _default3,
  _default4,
  _default5,
  _default6,
];
const features = {
  enableServer: true,
};
const __modelsContext = require.context(
  "../../model",
  false,
  /\\.[mc]?[jt]sx?$/
);
const models = __modelsContext.keys().reduce((res, filename) => {
  const pattern = __getModelPattern(filename);
  return {
    ...res,
    [pattern]: __modelsContext(filename).default,
  };
}, {});
function __getModelPattern() {
  const modelFilename = arguments[0];
  const MODEL_PATTERN_REGEX = /^[a-zA-Z0-9$_*.]+$/;
  let pattern = modelFilename;
  if (/\\*/.test(pattern)) {
    throw Error(
      "[models] Instead of '*' in model filename use '[id]'. Got: " +
        modelFilename
    );
  }
  pattern = pattern.replace(/\\[[^\\]]*\\]/g, "*");
  pattern = pattern.replace(/^.+[\\\\/]/, "");
  pattern = pattern.replace(/\\.[^.]+$/, "");
  if (!MODEL_PATTERN_REGEX.test(pattern)) {
    throw Error(
      "[models] Invalid model filename pattern: " +
        modelFilename +
        "\\n" +
        "It has to comply with the following regex: " +
        MODEL_PATTERN_REGEX.toString() +
        " with '[id]' instead of '*'"
    );
  }
  if (pattern === "index") pattern = "";
  return pattern;
}
config.features = features;
registry.init(config, {
  plugins,
  models,
});
const x = "xxx";
dummy(x);
export default () => {};


`;

exports[`@startupjs/babel-plugin-startupjs-plugins Processes files with a magic import: Processes files with a magic import 1`] = `

import { registry } from 'startupjs/registry'
import config from './startupjs.config.virtual.js'
import models from './startupjs.models.virtual.js'
import features from './startupjs.features.virtual.js'
import plugins from './startupjs.plugins.virtual.js'
import dummy from '@dummy/dummy'

config.features = features
registry.init(config, { plugins, models })

const x = 'xxx'
dummy(x)

export default () => {}

      ↓ ↓ ↓ ↓ ↓ ↓

import { registry } from "startupjs/registry";
import config from "../../startupjs.config.js";
import dummy from "@dummy/dummy";
import _default from "../../model/_session.games.[id].js";
import _default2 from "../../model/_session.games.js";
import _default3 from "../../model/index.js";
import _default4 from "../../model/users.[id].js";
import _default5 from "../../model/users.js";
import _default6 from "@startupjs/server/plugins/clientSession.plugin";
import _default7 from "@startupjs/ui/plugin";
import _default8 from "startupjs/plugins/cssMediaUpdater.plugin";
import _default9 from "module-1/plugin";
import _default10 from "module-1-plugin/thePlugin.plugin";
import _default11 from "../../dummyPlugin.plugin.js";
const plugins = [
  _default6,
  _default7,
  _default8,
  _default9,
  _default10,
  _default11,
];
const features = {
  enableServer: true,
};
const models = {
  "_session.games.*": _default,
  "_session.games": _default2,
  "": _default3,
  "users.*": _default4,
  users: _default5,
};
config.features = features;
registry.init(config, {
  plugins,
  models,
});
const x = "xxx";
dummy(x);
export default () => {};


`;

exports[`@startupjs/babel-plugin-startupjs-plugins Test sample file from fixtures which loads config: Test sample file from fixtures which loads config 1`] = `

import { registry } from 'startupjs/registry'
import config from './startupjs.config.virtual.js'
import models from './startupjs.models.virtual.js'
import features from './startupjs.features.virtual.js'
import plugins from './startupjs.plugins.virtual.js'

config.features = features
registry.init(config, { plugins, models })

      ↓ ↓ ↓ ↓ ↓ ↓

import { registry } from "startupjs/registry";
import config from "../../startupjs.config.js";
import _default from "../../model/_session.games.[id].js";
import _default2 from "../../model/_session.games.js";
import _default3 from "../../model/index.js";
import _default4 from "../../model/users.[id].js";
import _default5 from "../../model/users.js";
import _default6 from "@startupjs/server/plugins/clientSession.plugin";
import _default7 from "@startupjs/ui/plugin";
import _default8 from "startupjs/plugins/cssMediaUpdater.plugin";
import _default9 from "module-1/plugin";
import _default10 from "module-1-plugin/thePlugin.plugin";
import _default11 from "../../dummyPlugin.plugin.js";
const plugins = [
  _default6,
  _default7,
  _default8,
  _default9,
  _default10,
  _default11,
];
const features = {
  enableServer: true,
};
const models = {
  "_session.games.*": _default,
  "_session.games": _default2,
  "": _default3,
  "users.*": _default4,
  users: _default5,
};
config.features = features;
registry.init(config, {
  plugins,
  models,
});


`;
